### PR TITLE
Revert "remove redundant wfi fetch"

### PR DESCRIPTION
**Pull request by honnix**
Reverts spotify/styx#466

As we discussed, we might want to consider bring it back. @danielnorberg

### CODE REVIEW

**Comment by honnix** [comment]
cc @NatashaL

File: styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java [hunk 1]
```diff
@@ -472,20 +472,22 @@ void pollPods() {
...
       final Optional<WorkflowInstance> workflowInstance = readPodWorkflowInstance(pod);
       if (!workflowInstance.isPresent()) {
         continue;
       }
-      final RunState runState = activeStates.get(workflowInstance.get());
-      if (runState != null && isPodRunState(pod, runState)) {
-        emitPodEvents(Watcher.Action.MODIFIED, pod, runState);
+      // do an extra lookup to get latest state
+      // this is crucial because after fetching all states in datastore, some states might
+      // have transitioned already
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java]: We might even observe pods whose states where created after fetching the states.

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java]: Yeah good catch. Could you help fix this? I’m afk now. :)

**Comment by danielnorberg** [comment]
Looks to me like there's additional race conditions in this polling code. Maybe we should spend some time on it.

**Comment by honnix** [comment]
I'm attempted to revert the change to mitigate current issue. Seems quite some cases in the past week. The other race conditions don't seem to have bad impact.

**Comment by danielnorberg** [comment]
👍

**Pull request merged**